elastic/kibana
# [Response Ops][Alerting] Update FAAD `AlertsClient` to support AAD payload

Resolves https://github.com/elastic/kibana/issues/156443, https://github.com/elastic/kibana/issues/156445

## Summary

- Updates `AlertsClient` with `create` API that allows rule executors to report alerts with AAD payload, along with the values for `actionGroup`, `context` and `state`. This proxies the `LegacyAlertsClient` to create alerts via the alerts factory but also saves the reported payload
- When the alert doc is bulk written at the end of rule execution, the AAD payload (if specified) is included in the alert document
- Deprecates the alert factory that is passed into the rule executors, but this PR does not remove or replace usages of the alert factory
- Expose `AlertsClient` services to the rule executors. Note that this PR does not migrate any rule type to use this service.

This PR does not opt any rule types into writing the AAD payload or using the AlertsClient API but updates the AAD functional test to do so. To test it out with the ES query rule type, use the following commit: https://github.com/elastic/kibana/pull/158404/commits/1b1e139f805d98b7d4425f5de5124788e3f75d96

## Followup issues
- This PR does not add a recovery API to the FAAD AlertsClient so alerts reported via the new alerts client currently do not have a way of specifying recovered payload.

## To Verify
- Verify that rule registry rule types still work as expected
- Verify that non rule-registry rule types still work as expected
- Check out [this commit](https://github.com/elastic/kibana/pull/158404/commits/1b1e139f805d98b7d4425f5de5124788e3f75d96) which onboards the ES query rule type onto FAAD. Create an ES query rule that alerts and then recovers and verify that the alert documents look as expected. Alternatively, you can modify your own rule type to register with FAAD and write alerts and verify that the alert documents look as expected.

### Checklist
